Makefile

```diff
@@ -1,48 +1,50 @@
 .PHONY: dev lint complex coverage pre-commit yapf sort deploy destroy deps unit infra-tests integration e2e coverage-tests docs lint-docs build
+CWD := $(shell pwd)
 
-
-
+.ONESHELL:  # run all commands in a single shell, ensuring it runs within a local virtual env
 dev:
+	python3 -m venv .venv
+	. .venv/bin/activate
 	pip install --upgrade pip pre-commit poetry
 	pre-commit install
 	poetry config --local virtualenvs.in-project true
 	poetry install
+	npm ci
```

Comment:
this lines add a new folder:
node_modules, dont we need to add it to the gitignore and it also fails my yapf. so weird that it works in the pipeline

Reply:
I had added it: https://github.com/ran-isenberg/serverless-python-demo/blob/6a4af44d746cf9e03e738157cad5dc5ae25e4d8c/.gitignore#L254

For yapf, lemme learn more about it - I expected it to only look for Python files. I ran locally it was fine.

Reply:
Found the YAPF docs and they support ignoring files within `pyproject.toml` - working now.

```toml
[tool.yapfignore]
ignore_patterns = [".git", ".venv", ".build", "cdk.out", "node_modules"]
```